x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elser/ElserMlNodeServiceSettings.java

```diff
@@ -61,21 +62,48 @@ public static ElserMlNodeServiceSettings fromMap(Map<String, Object> map) {
             validationException.addValidationError(mustBeAPositiveNumberError(NUM_THREADS, numThreads));
         }
 
+        String version = MapParsingUtils.removeAsType(map, MODEL_VERSION, String.class);
+        if (version != null && ElserMlNodeService.VALID_ELSER_MODELS.contains(version) == false) {
+            validationException.addValidationError("unknown ELSER model version [" + version + "]");
+        }
+
         if (validationException.validationErrors().isEmpty() == false) {
             throw validationException;
         }
 
-        return new ElserMlNodeServiceSettings(numAllocations, numThreads);
+        var builder = new Builder();
+        builder.setNumAllocations(numAllocations);
+        builder.setNumThreads(numThreads);
+        builder.setModelVariant(version);
+        return builder;
     }
 
-    public ElserMlNodeServiceSettings(int numAllocations, int numThreads) {
+    public ElserMlNodeServiceSettings(int numAllocations, int numThreads, String variant) {
         this.numAllocations = numAllocations;
         this.numThreads = numThreads;
+        this.modelVariant = Objects.requireNonNull(variant);
     }
 
     public ElserMlNodeServiceSettings(StreamInput in) throws IOException {
         numAllocations = in.readVInt();
         numThreads = in.readVInt();
+        if (transportVersionIsCompatibleWithElserModelVersion(in.getTransportVersion())) {
+            modelVariant = in.readString();
+        } else {
+            modelVariant = ElserMlNodeService.ELSER_V2_MODEL;
```

Comment:
I think it _would_ be better to stick to `ELSER_V1_MODEL` in the mixed version cluster where some of the nodes are pre-8.11.0. The `ELSER_V2_MODEL` states that it requires 8.11.0 (because it uses the platform specific config functionality). So it might not install in a mixed version cluster where some nodes pre-date 8.11.0.

Reply:
Same as on the other PR, this code should never be decoding anything from a pre-8.11 node as this class doesn't exist in pre-8.11 code. So what goes here doesn't matter.